### DISCUSSION TITLE
Support for importExternalTexture?

### DISCUSSION THREAD

**Discussion by bes** [gfx-rs/wgpu#4052] [Q&A]
Hi!

I'm eager to start using `wgpu`, but for my use-case `importExternalTexture` is the most important missing piece (specifically for web/WebGL2 & iOS/metal).

For web I want to grab a texture from a `HTMLVideoElement` and from iOS I can get a `MTLTexture`.

What is the work needed to reach `importExternalTexture` for web, and using an external `MTLTexture` for iOS?

Can I help / contribute in some way?

**Comment by davidgarywood** [comment]
I too would love this !
